internal/restorable/images_js.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 The Ebitengine Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package restorable
+
+import (
+	"archive/zip"
+	"bytes"
+	"image"
+	"syscall/js"
+
+	"github.com/hajimehoshi/ebiten/v2/internal/graphicsdriver"
+)
+
+// DumpImages dumps all the current images to the specified directory.
+//
+// This is for testing usage.
+func DumpImages(graphicsDriver graphicsdriver.Graphics, dir string) error {
+	buf := &bytes.Buffer{}
+	zw := zip.NewWriter(buf)
+
+	for img := range theImages.images {
+		f, err := zw.Create(img.image.DumpName("*.png"))
+		if err != nil {
+			return err
+		}
+
+		if err := img.image.DumpTo(f, graphicsDriver, false, image.Rect(0, 0, img.width, img.height)); err != nil {
+			return err
+		}
+	}
+
+	zw.Close()
+
+	global := js.Global()
+
+	jsData := global.Get("Uint8Array").New(buf.Len())
+	js.CopyBytesToJS(jsData, buf.Bytes())
+
+	a := global.Get("document").Call("createElement", "a")
+	blob := global.Get("Blob").New(
+		[]interface{}{jsData},
+		map[string]interface{}{"type": "archive/zip"},
+	)
+	a.Set("href", global.Get("URL").Call("createObjectURL", blob))
+	a.Set("download", dir+".zip")
+	a.Call("click")
+
+	return nil
```

Comment:
Can we move this (and notjs's) implementation of DumpImages to `graphicscommand`? `graphicscommand.DumpImages` would accept `[]*graphicscommand.Image`.

1. We don't have to expose `DumpName`
2. We can reuse the hack to download something from browser by `<a>`.

Reply:
fantastic idea 👍 I'll do that

Reply:
How should I invoke `DumpImages` from atlas? Should I pull images from `theBackends`?

Reply:
Hmm, `theBackends` wouldn't work, since we don't have access to `(*restorable.Image).image` from atlas

Reply:
`restorable` already knows the images to dump so you don't have to modify `atlas`, right?

Reply:
oh gotcha, my bad. my brain isn't awake yet haha